PythonScripts/Movement/MoveImpl.py

```diff
@@ -46,221 +64,188 @@ def get_origin(self):
         # 8. opens Hand
         self._grip_open()
         # 9. Moves arm up
-        pos_to_value[2] += constants.DELTA_ABOVE_OBJ
-        self._move_arm(pos_to_value, rotation={'y': -90, 'x': 0, 'z': mapper.get_hand_rotation(pos_to_enum)})
-        pos_to_value[2] -= constants.DELTA_ABOVE_OBJ
+        position_to_coordinates[2] += constants.DELTA_HEIGHT
+        self._move_arm(position_to_coordinates,
+                       rotation={'y': -90, 'x': 0, 'z': mapper.get_hand_rotation(position_to)})
+        position_to_coordinates[2] -= constants.DELTA_HEIGHT
         self.move_head()
         # 10. Moves arm back to a save position
         self._grip_close()
         self.set_arm_to_right_angle_position()
         # 11. Moving arm to the origin coordinate, so that it does not block the view
-        self._move_arm(self.addlists(self.origin_coordinate, [0,-0.085,0.10]), rotation={'y': -90, 'x': 0, 'z': 0})
+        self._move_arm(add_lists(self.origin, [0, -0.085, 0.10]), rotation={'y': -90, 'x': 0, 'z': 0})
 
         # Setting arm to compliant mode and lowering smoothly for preventing damaging
-        #self.deactivate_right_arm()
+        # self.deactivate_right_arm()
         # head back to default and setting head to compliant mode
-        self.move_head(constants.HEAD_LOOK_AHEAD)
-        
-        pass
+        self.move_head(constants.HEAD_LOOK_FRONT)
 
     def _move_arm(self, pos_to: list, rotation: dict):
         """
         Moving arm to Position
 
         """
-        # Only adjust the rot_direction [x, y, z] and the rot_axis of type deg and then the arm movement
-        # should be precise
         target_kinematic = self.kinematic_model_helper.get_kinematic_move(pose=pos_to, rotation=rotation)
-
         joint_pos_A = self.reachy.r_arm.inverse_kinematics(target_kinematic)
         goto({joint: pos for joint, pos in zip(self.reachy.r_arm.joints.values(), joint_pos_A)}, duration=2.0)
-        pass
 
     def _change_grip_force(self, force):
         self.POS_GRIPPER[self.reachy.r_arm.r_gripper] = force
-        #print("current force:", self.POS_GRIPPER[self.reachy.r_arm.r_gripper])
-        pass
+        # print("current force:", self.POS_GRIPPER[self.reachy.r_arm.r_gripper])
 
     def _grip_open(self):
         """
-        opens grip completly
+        opens grip completely
         """
-        # Open grip completly
-        self._change_grip_force(-60)
+        self._change_grip_force(constants.GRIPPER_OPEN_FULL)
         goto(goal_positions=self.POS_GRIPPER, duration=1.0, interpolation_mode=InterpolationMode.MINIMUM_JERK)
-        pass
 
     def _grip_close(self):
         """
         closes grip until is_holding is true
         """
-        # Closes grip
-        # TODO: CLOSE until _is_holding
-        self._change_grip_force(5)
+        MAX_TIME_PASSED = 3.0
+        force = 1
+        start = time.time()
+        while not self._is_holding():
```

Comment:
The While loop actually never closes the hand!
The Function of closing hand is called later